src/modules/Dashboard/features/Applet/Popups/UpgradeAccountPopup/UpgradeAccountPopup.tsx

```diff
@@ -0,0 +1,148 @@
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import { Modal, Spinner, SubmitBtnVariant } from 'shared/components';
+import {
+  StyledBodyLarge,
+  StyledErrorText,
+  StyledFlexColumn,
+  StyledModalWrapper,
+} from 'shared/styles';
+import { useFormError } from 'modules/Dashboard/hooks';
+import { Mixpanel, getErrorMessage } from 'shared/utils';
+import { Languages, postSubjectInvitationApi } from 'api';
+import { useAppDispatch } from 'redux/store';
+import { useAsync } from 'shared/hooks';
+import { banners } from 'redux/modules';
+import { AccountType } from 'modules/Dashboard/types';
+import { ParticipantSnippet } from 'modules/Dashboard/components';
+
+import { RESPONDENT_ALREADY_INVITED } from './UpgradeAccountPopup.const';
+import { UpgradeAccountPopupSchema } from './UpgradeAccountPopup.schema';
+import {
+  UpgradeAccountPopupProps,
+  UpgradeAccountFormValues,
+  Fields,
+} from './UpgradeAccountPopup.types';
+import { UpgradeAccountForm } from './UpgradeAccountForm';
+
+export const UpgradeAccountPopup = ({
+  popupVisible,
+  appletId,
+  subjectId,
+  secretId,
+  nickname,
+  tag,
+  onClose,
+  'data-testid': dataTestid,
+}: UpgradeAccountPopupProps) => {
+  const { t, i18n } = useTranslation('app');
+  const defaultValues = {
+    email: '',
+    language: i18n.language as Languages,
+  };
+  const {
+    handleSubmit,
+    control,
+    reset,
+    setError,
+    formState: { isValid, isDirty },
+  } = useForm<UpgradeAccountFormValues>({
+    resolver: yupResolver(UpgradeAccountPopupSchema()),
+    defaultValues,
+    mode: 'all',
+    delayError: 1000,
```

Comment:
how about a constant value for it? For ex., for input fields we used `CHANGE_DEBOUNCE_VALUE` maybe it applicable in your case as well or does it make sense to introduce a new constant for such form setting?

Reply:
I agree, this should be a global constant as it's now used in multiple places, and this aspect of UX should be consistent. Thanks!